File: java/src/main/java/edu/ucdavis/fiehnlab/spectra/hash/core/Spectrum.java
package edu.ucdavis.fiehnlab.spectra.hash.core;

import edu.ucdavis.fiehnlab.spectra.hash.core.types.SpectraType;
import edu.ucdavis.fiehnlab.spectra.hash.core.types.Ion;

import java.util.List;
import java.util.Map;

/**
 * describes a spectrum for a splash
 */
public interface Spectrum {

    /**
     * ion of a spectra
     * @return
     */
    public List<Ion> getIons();

    /**
     * convmerts the spectrum to a relative spectra
     * @return
     * @param scale
     */
    Spectrum toRelative(int scale);

    /**
     *
     * @return
     */
    String getOrigin();

    /**
     * what kind of a spectra do we have
     * @return
     */
    public SpectraType getType();
}

File: scala/src/main/scala/edu/ucdavis/fiehnlab/splash/scala/Spectrum.scala
package jp.riken.mirt.splash

import edu.ucdavis.fiehnlab.spectra.hash.core
import jp.riken.mirt.splash.JavaConversions._
import scala.collection.JavaConversions._

trait Spectrum extends core.Spectrum {
  lazy val splashIt: String = { 
    val ions: Seq[Ion] = getIons().map(ion => Ion(ion.getMass, ion.getIntensity)) 
    SplashFactory.create().splashIt(ions)
  }
}

